quarkchain/cluster/shard_state.py

```diff
@@ -1995,4 +2015,22 @@ def get_total_balance(
         limit: int,
         starter: Optional[bytes] = None,
     ) -> Tuple[int, bytes]:
-        return 42, b""
+        """
+        The starter address should be exclusive in the iteration.
+        """
+        evm_state = self._get_evm_state_from_hash(block_hash)
+        # the secure trie stored the pair of hash to key
+        # the trie stored the pair of hash to node
+        trie = evm_state.trie.trie
+        # hash the address to get the paired key, then get next key for exclusive search
+        key = trie.next(bytes(32) if not starter else utils.sha3_256(starter))
+        total = 0
+        ret = None
+        while limit > 0 and key is not None:
+            addr = evm_state.trie.db.get(key)
```

Comment:
how to handle if `key` is not present in trie's db?

Reply:
do we have a test case that's calling the function with non-existing address?

Reply:
If the key is not present in trie's db, the get function will return `None`, and an error will be raised in this function. A testcase is also added.